app/app/components/Profile.jsx

```diff
@@ -1,28 +1,30 @@
-import React, { useState, useEffect } from "react";
-import axios from "axios";
+import { useState, useEffect, useCallback } from "react";
 import Button from "./Button";
+import { Box } from "@mui/material";
 import EditProfile from "./EditProfile";
+import GoalsWidget from "./GoalsWidget";
+import Bookshelf from "./Bookshelf";
+import { makeRequest } from "../utils/makeRequest";
 import styles from "./Profile.module.css";
 
-const Profile = ({ userId, booksReadCount }) => {
+const Profile = ({ booksReadCount }) => {
   const [userData, setUserData] = useState(null);
-  const [isModalOpen, setModalOpen] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
   const [favoriteGenre, setFavoriteGenre] = useState(null);
+  const [activeGoal, setActiveGoal] = useState(null);
+  const [bookShelf, setBookShelf] = useState({
+    read: [],
+    currentlyReading: [],
+    wishToRead: [],
+  });
 
   useEffect(() => {
```

Comment:
first of all, I would suggest to spread the fetching data into a separate effect each. It is easy to get things bloated and they are independent from each other anyway.